service/controller/resource/deployment/template/main.json

```diff
@@ -732,6 +732,7 @@
     {
       "apiVersion":"2016-09-01",
       "name":"virtual_network_setup",
+      "condition":"[equals(parameters('initialProvisioning'), 'Yes')]",
```

Comment:
This is the most relevant change. With this change, we only submit the Virtual Network deployment if we didn't find the three expected subnets (vpn, masters, workers). If we find them, this deployment is not submitted. Meaning we wouldn't be able to update the CIDR of an existing virtual network using the ARM deployment.

Reply:
So if we ever need to modify VNet somehow, we cannot use this ARM template anymore and we will have to do it with SDK?

Reply:
Yes. The deployment would not be submitted since it already exists, and changes done using the SDK would be kept.

Reply:
I changed the code: we will only submit the virtual network deployment if the `cluster-main-template` deployment is not found.

Reply:
This is then a temporary change IIUC, as it is probably just a matter of time when we will need to change something around VNet again, so in the future will be doing that from code.

Reply:
Could this be changed to only consider vpn & masters. I just had a cluster without workers and technically the CP is okayish (I know there aren't resources for some deployments but that's different story).

Reply:
Stupid me. Wrote & submitted before reading till the end. All good.

Reply:
...but where's the related code?

Reply:
What related code? [This](https://github.com/giantswarm/azure-operator/blob/ae43c102dae983790a716a2110d1690482e0c267/service/controller/resource/deployment/resource.go#L115-L118)?